pallets/api/src/nonfungibles/mod.rs

```diff
@@ -0,0 +1,621 @@
+		#[pallet::call_index(16)]
+		#[pallet::weight(NftsWeightInfoOf::<T>::approve_item_attributes())]
+		pub fn approve_item_attributes(
+			origin: OriginFor<T>,
+			collection: CollectionIdOf<T>,
+			item: ItemIdOf<T>,
+			delegate: AccountIdOf<T>,
+		) -> DispatchResult {
+			NftsOf::<T>::approve_item_attributes(
+				origin,
+				collection,
+				item,
+				T::Lookup::unlookup(delegate),
+			)
+		}
+
+		/// Cancel the previously provided approval to change item's attributes.
+		/// All the previously set attributes by the `delegate` will be removed.
+		///
+		/// # Parameters
+		/// - `collection` - The collection that the item is contained within.
+		/// - `item` - The item that holds attributes.
+		/// - `delegate` - The previously approved account to remove.
+		/// - `witness` - A witness data to cancel attributes approval operation.
+		#[pallet::call_index(17)]
+		#[pallet::weight(NftsWeightInfoOf::<T>::cancel_item_attributes_approval(witness.account_attributes))]
+		pub fn cancel_item_attributes_approval(
+			origin: OriginFor<T>,
+			collection: CollectionIdOf<T>,
+			item: ItemIdOf<T>,
+			delegate: AccountIdOf<T>,
+			witness: CancelAttributesApprovalWitness,
+		) -> DispatchResult {
+			NftsOf::<T>::cancel_item_attributes_approval(
+				origin,
+				collection,
+				item,
+				T::Lookup::unlookup(delegate),
+				witness,
+			)
+		}
+
+		/// Set the maximum number of items a collection could have.
+		///
+		/// # Parameters
+		/// - `collection` - The collection identifier.
+		/// - `max_supply` - The maximum number of items a collection could have.
+		#[pallet::call_index(18)]
+		#[pallet::weight(NftsWeightInfoOf::<T>::set_collection_max_supply())]
+		pub fn set_max_supply(
+			origin: OriginFor<T>,
+			collection: CollectionIdOf<T>,
+			max_supply: u32,
+		) -> DispatchResult {
+			NftsOf::<T>::set_collection_max_supply(origin, collection, max_supply)
+		}
+
+		/// Mint an item of a particular collection.
+		///
+		/// # Parameters
+		/// - `to` - Account into which the item will be minted.
+		/// - `collection` - The collection of the item to mint.
+		/// - `item` - An identifier of the new item.
+		/// - `witness` - When the mint type is `HolderOf(collection_id)`, then the owned item_id
+		///   from that collection needs to be provided within the witness data object. If the mint
+		///   price is set, then it should be additionally confirmed in the `witness`.
+		#[pallet::call_index(19)]
+		#[pallet::weight(NftsWeightInfoOf::<T>::mint())]
+		pub fn mint(
+			origin: OriginFor<T>,
+			to: AccountIdOf<T>,
+			collection: CollectionIdOf<T>,
+			item: ItemIdOf<T>,
+			witness: MintWitness<ItemIdOf<T>, ItemPriceOf<T>>,
+		) -> DispatchResult {
+			let owner = ensure_signed(origin.clone())?;
+			NftsOf::<T>::mint(origin, collection, item, T::Lookup::unlookup(to), Some(witness))?;
+			Self::deposit_event(Event::Transfer { collection, item, from: None, to: Some(owner) });
```

Comment:
Dont understand why this is not `to`:
```suggestion
			Self::deposit_event(Event::Transfer { collection, item, from: None, to });
```